### PR TITLE
YQ-2068 Introduce stream lookup input transform in passthrough(no lookup) mode

### PR DESCRIPTION
**Pull request by zverevgeny**
...

### CODE REVIEW

File: ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    {
+        Y_ABORT_UNLESS(Alloc);
+    }
+
+    void Bootstrap() {
+        //TODO implement me
+    }
+
+private:
+private: //IDqComputeActorAsyncInput
```
Comment (Darych) [inline ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp:54]: Do we need this comment?

File: ydb/library/yql/dq/type_ann/dq_type_ann.cpp [hunk 2]
```diff
@@ -995,6 +1023,10 @@ THolder<IGraphTransformer> CreateDqTypeAnnotationTransformer(TTypeAnnotationCont
             if (TDqCnMerge::Match(input.Get())) {
                 return AnnotateDqCnMerge(input, ctx);
             }
+            
+            if (TDqCnStreamLookup::Match(input.Get())) {
```
[the remark this answers is not in the document]
Comment (zverevgeny) [reply ydb/library/yql/dq/type_ann/dq_type_ann.cpp]: fixed

File: ydb/library/yql/providers/dq/opt/physical_optimize.cpp [hunk 3]
```diff
@@ -204,6 +205,49 @@ class TDqsPhysicalOptProposalTransformer : public TOptimizeTransformerBase {
...
     }
 
+    TMaybeNode<TExprBase> RewriteStreamLookupJoin(TExprBase node, TExprContext& ctx) {
+        const auto join = node.Cast<TDqJoin>();
+        if (join.JoinAlgo().StringValue() != "StreamLookupJoin") {
+            return node;
+        }
+
+        const auto pos = node.Pos();
+        const auto left = join.LeftInput().Maybe<TDqConnection>(); //Maybe<TDqConnection>
```
[the remark this answers is not in the document]
Comment (zverevgeny) [reply ydb/library/yql/providers/dq/opt/physical_optimize.cpp]: done

File: ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp (hunk 1 above)
Comment (zverevgeny) [reply ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp:53]: It aims to group methods that implement this interface

**Review by Darych — DISMISSED** [review]

**Review by Darych — APPROVED** [review]

**Pull request merged**